### PR TITLE
Fix #9283: Description and tags are not getting updated for schema fields

### LINKED ISSUE #9283: Backend : Description and tags are not getting updated for schema fields.
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Backend

**Describe the bug**

Description and tags are not getting updated for schema fields.

**To Reproduce**
- Go to any topic entity
- Try to edit the description or tags of any schema field
- API returns the successful response but data is not getting saved into the DB.

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

Payload : 

```json
[
    {
        "op": "add",
        "path": "/messageSchema/schemaFields/2/children/0/children/1/children/0/children/1/description",
        "value": "Description."
    }
]
```

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :

I worked on the ..... because ...

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/util/EntityUtil.java [hunk 1]
```diff
@@ -375,6 +386,16 @@ public static String getColumnField(Table table, Column column, String columnFie
         : FullyQualifiedName.build("columns", localColumnName, columnField);
   }
 
+  /** Return schema field name of format "schemaFields".fieldName.fieldName */
+  public static String getSchemaField(Topic topic, Field field, String fieldName) {
+    // Remove table FQN from column FQN to get the local name
+    String localColumnName =
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/EntityUtil.java:392]: Comments still refer to table and column

File: openmetadata-service/src/main/java/org/openmetadata/service/util/EntityUtil.java [hunk 2]
```diff
@@ -146,6 +149,14 @@ public final class EntityUtil {
 
   public static final BiPredicate<Rule, Rule> ruleMatch = (ref1, ref2) -> ref1.getName().equals(ref2.getName());
 
+  public static final BiPredicate<Field, Field> filedMatch =
+      (field1, field2) ->
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/EntityUtil.java:153]: Typo. Should be FieldMatch?

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java [hunk 3]
```diff
@@ -257,5 +270,85 @@ private void updateCleanupPolicies(Topic original, Topic updated) throws JsonPro
...
+        return;
+      }
+      String field = getSchemaField(original, origField, FIELD_DISPLAY_NAME);
+      recordChange(field, origField.getDescription(), updatedField.getDescription());
+    }
+
+    private void updateFieldDisplayName(Field origField, Field updatedField) throws JsonProcessingException {
+      if (operation.isPut() && !nullOrEmpty(origField.getDescription()) && updatedByBot()) {
+        // Revert the non-empty task description if being updated by a bot
+        updatedField.setDisplayName(origField.getDisplayName());
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java:346]: task description seems to cut and past error in the comment

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java [hunk 4]
```diff
@@ -257,5 +270,85 @@ private void updateCleanupPolicies(Topic original, Topic updated) throws JsonPro
...
+          updateSchemaFields(childrenFieldName, stored.getChildren(), updated.getChildren(), fieldMatch);
+        }
+      }
+
+      majorVersionChange = majorVersionChange || !deletedFields.isEmpty();
+    }
+
+    private void updateFieldDescription(Field origField, Field updatedField) throws JsonProcessingException {
+      if (operation.isPut() && !nullOrEmpty(origField.getDescription()) && updatedByBot()) {
+        // Revert the non-empty task description if being updated by a bot
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java:335]: task description in the comment

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java [hunk 5]
```diff
@@ -257,5 +270,85 @@ private void updateCleanupPolicies(Topic original, Topic updated) throws JsonPro
...
+      // Delete tags related to deleted columns
+      deletedFields.forEach(
+          deleted -> daoCollection.tagUsageDAO().deleteTagsByTarget(deleted.getFullyQualifiedName()));
+
+      // Add tags related to newly added columns
+      for (Field added : addedFields) {
+        applyTags(added.getTags(), added.getFullyQualifiedName());
+      }
+
+      // Carry forward the user generated metadata from existing columns to new columns
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java:308]: columns in the comment should be fields

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java [hunk 6]
```diff
@@ -257,5 +270,85 @@ private void updateCleanupPolicies(Topic original, Topic updated) throws JsonPro
...
+            addedField.setTags(deleted.getTags());
+          }
+        }
+      }
+
+      // Delete tags related to deleted columns
+      deletedFields.forEach(
+          deleted -> daoCollection.tagUsageDAO().deleteTagsByTarget(deleted.getFullyQualifiedName()));
+
+      // Add tags related to newly added columns
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java:303]: columns in the comment should be fields?

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java [hunk 7]
```diff
@@ -257,5 +270,85 @@ private void updateCleanupPolicies(Topic original, Topic updated) throws JsonPro
...
+    private void updateSchemaFields(
+        String fieldName,
+        List<Field> origFields,
+        List<Field> updatedFields,
+        BiPredicate<Field, Field> fieldMatch)
+        throws IOException {
+      List<Field> deletedFields = new ArrayList<>();
+      List<Field> addedFields = new ArrayList<>();
+      recordListChange(fieldName, origFields, updatedFields, addedFields, deletedFields, fieldMatch);
+      // carry forward tags and description if deletedColumns matches added column
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TopicRepository.java:283]: columns to fields?

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**